library/time-library/src/main/java/uk/gov/gchq/gaffer/time/function/FilterTimestampSetByTimeRange.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016-2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.time.function;
+
+import uk.gov.gchq.gaffer.time.RBMBackedTimestampSet;
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import java.time.Instant;
+
+/**
+ * A {@code FilterTimestampSetByTimeRange} is a {@link uk.gov.gchq.koryphe.function.KorypheFunction} that takes in a
+ * {@link RBMBackedTimestampSet} and filters the internal timestamps by a start time and end time.
+ */
+@Since("1.9.0")
+@Summary("Filters a RBMBackedTimestampSet by a time range")
+public class FilterTimestampSetByTimeRange extends KorypheFunction<RBMBackedTimestampSet, RBMBackedTimestampSet> {
```

Comment:
This should probably be called something like Mask rather than Filter. To me, filter implied you were creating a new object rather than mutating the original object (more like a transform function).

Reply:
Good point